File: mobile/lib/models/annee_scolaire.dart
class AnneeScolaire {
  final int id;
  final String libelle;
  final String statut; // en_cours | en_cloture | cloturee
  final String dateDebut;
  final String dateFin;

  const AnneeScolaire({
    required this.id,
    required this.libelle,
    required this.statut,
    required this.dateDebut,
    required this.dateFin,
  });

  bool get estCloturee => statut == 'cloturee';
  bool get estEnCours  => statut == 'en_cours';

  factory AnneeScolaire.fromJson(Map<String, dynamic> json) {
    return AnneeScolaire(
      id:        json['id'] as int,
      libelle:   json['libelle'] as String,
      statut:    json['statut'] as String,
      dateDebut: json['date_debut'] as String,
      dateFin:   json['date_fin'] as String,
    );
  }
}

File: back/app/Models/AnneeScolaire.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class AnneeScolaire extends Model
{
    protected $table = 'annees_scolaires';

    protected $fillable = ['libelle', 'date_debut', 'date_fin', 'statut'];

    protected function casts(): array
    {
        return [
            'date_debut' => 'date',
            'date_fin'   => 'date',
        ];
    }

    // ── Scopes ────────────────────────────────────────────────────────────────

    public function scopeEnCours($query)
    {
        return $query->where('statut', 'en_cours');
    }

    public function scopeCloturee($query)
    {
        return $query->where('statut', 'cloturee');
    }

    // ── Relations ─────────────────────────────────────────────────────────────

    public function periodes()
    {
        return $this->hasMany(Periodes::class);
    }

    public function passageClasses()
    {
        return $this->hasMany(PassageClasse::class);
    }

    // ── Helpers ───────────────────────────────────────────────────────────────

    public static function courante(): ?self
    {
        return static::where('statut', 'en_cours')->latest()->first();
    }

    public function estModifiable(): bool
    {
        return $this->statut === 'en_cours';
    }

    public function estEnCloture(): bool
    {
        return $this->statut === 'en_cloture';
    }

    public function estCloturee(): bool
    {
        return $this->statut === 'cloturee';
    }
}
